pkg/cloudflare/decisions-sync-worker/src/core/decision-fetcher.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * CrowdSec LAPI Decision Fetcher
+ * Fetches security decisions from CrowdSec LAPI or BLaaS endpoint
+ * Based on the Node.js bouncer implementation pattern
+ */
+
+import logger from '../utils/logger.js';
+
+const USER_AGENT = 'cloudflare-worker-bouncer/v1.0.0';
+const WARMUP_KEY = 'WARMUP';
+const SUPPORTED_SCOPES = ['ip', 'range', 'as', 'country'];
+
+/**
+ * Check if this is the first fetch by looking for the WARMUP flag in KV
+ * @param {KVNamespace} kvNamespace - Cloudflare KV namespace
+ * @returns {Promise<boolean>} True if this is the first fetch
+ */
+export async function isFirstFetch(kvNamespace) {
+	const warmupFlag = await kvNamespace.get(WARMUP_KEY);
+	return !warmupFlag;
+}
+
+/**
+ * Mark the cache as warmed up (first fetch completed)
+ * @param {KVNamespace} kvNamespace - Cloudflare KV namespace
+ */
+export async function markAsWarmed(kvNamespace) {
+	await kvNamespace.put(WARMUP_KEY, 'true');
```

Comment:
Maybe renamed it WARMED_UP. When i first read it I thought it meant the opposite

Reply:
That would be more correct written : WARMED_UP